### PR TITLE
Fix #5129: Use system-dependent path separator

### LINKED ISSUE #5129: dotty.tools.dotc.CompilationTests fail on Windows
The issue is most likely related to different class path separator.
```
Testing tests\run\virtpatmat_valdef.scala
java.nio.file.InvalidPathException: Illegal char <:> at index 138: D:\home\projects\Scala\Dotty\git\library\..\out\bootstrap\dotty-library-bootstrapped\scala-0.10\dotty-library_0.10-0.10.0-bin-SNAPSHOT.jar:D:\home\projects\Scala\Dotty\git\out\runTestFromTasty\run\weakconform
	at sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at sun.nio.fs.WindowsPath.parse(WindowsPath.java:94)
	at sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:255)
	at java.nio.file.Paths.get(Paths.java:84)
	at dotty.tools.io.Directory$.apply(Directory.scala:32)
	at dotty.tools.io.AbstractFile$.getDirectory(AbstractFile.scala:25)
	at dotty.tools.dotc.classpath.ClassPathFactory.$anonfun$classesInPathImpl$1(ClassPathFactory.scala:58)
	at scala.collection.immutable.List.flatMap(List.scala:335)
	at dotty.tools.dotc.classpath.ClassPathFactory.classesInPathImpl(ClassPathFactory.scala:55)
	at dotty.tools.dotc.classpath.ClassPathFactory.classesInExpandedPath(ClassPathFactory.scala:44)
	at dotty.tools.dotc.config.PathResolver$Calculated$.basis(PathResolver.scala:234)
	at dotty.tools.dotc.config.PathResolver$Calculated$.containers$lzycompute(PathResolver.scala:238)
	at dotty.tools.dotc.config.PathResolver$Calculated$.containers(PathResolver.scala:238)
	at dotty.tools.dotc.config.PathResolver.containers(PathResolver.scala:261)
	at dotty.tools.dotc.config.PathResolver.result$lzycompute(PathResolver.scala:264)
	at dotty.tools.dotc.config.PathResolver.result(PathResolver.scala:263)
	at dotty.tools.dotc.config.JavaPlatform.classPath(JavaPlatform.scala:18)
	at dotty.tools.dotc.config.JavaPlatform.rootLoader(JavaPlatform.scala:38)
	at dotty.tools.dotc.core.Contexts$ContextBase.rootLoader(Contexts.scala:602)
```
I'm trying to fix it but the problem is that it's hard to do that w/o debug and I can't debug it from IJ Idea.
It either couldn't find test classes or fails for some other reasons.

Does someone know how to run/debug such tests from IJ Idea?

### PR DESCRIPTION
**Pull request by melekhove**

### CODE REVIEW

File: compiler/test/dotty/tools/vulpix/TestFlags.scala [hunk 1]
```diff
@@ -12,10 +12,10 @@ final case class TestFlags(
     TestFlags(defaultClassPath, runClassPath, options diff flags)
 
   def withClasspath(classPath: String): TestFlags =
-    TestFlags(s"$defaultClassPath:$classPath", runClassPath, options)
+    TestFlags(s"$defaultClassPath${sys.props("path.separator")}$classPath", runClassPath, options)
```
Comment (allanrenucci) [inline compiler/test/dotty/tools/vulpix/TestFlags.scala]: Isn't `java.io.File.pathSeparator` working?

File: compiler/test/dotty/tools/vulpix/TestFlags.scala [hunk 2]
```diff
@@ -12,10 +12,10 @@ final case class TestFlags(
     TestFlags(defaultClassPath, runClassPath, options diff flags)
 
   def withClasspath(classPath: String): TestFlags =
-    TestFlags(s"$defaultClassPath:$classPath", runClassPath, options)
+    TestFlags(s"$defaultClassPath${java.io.File.pathSeparator}$classPath", runClassPath, options)
 
   def withRunClasspath(classPath: String): TestFlags =
-    TestFlags(defaultClassPath, s"$runClassPath:$classPath", options)
+    TestFlags(defaultClassPath, s"$runClassPath${java.io.File.pathSeparator}$classPath", options)
```
Comment (Blaisorblade) [inline compiler/test/dotty/tools/vulpix/TestFlags.scala]: I’d prefer if you import File here too (maybe as JFile for consistency).

**Comment by Blaisorblade** [comment]
And the commit message should be updated (you’re not using a system property); and please say “fix #5129” rather than “fix issue #5129”.

**Comment by melekhove** [comment]
BTW java.nio.file.NoSuchFileException: ...i4947c\Aux.tasty in the log is the result of the #5148

**Comment by melekhove** [comment]
See also #5155 and #5157

**Comment by melekhove** [comment]
> And the commit message should be updated (you’re not using a system property); and please say “fix #5129” rather than “fix issue #5129”.
I've made all required changes

**Comment by OlivierBlanvillain** [comment]
@melekhove Merging, thanks!

**Pull request merged**